tests/s25Main/integration/testGamePlayer.cpp

```diff
@@ -62,8 +62,22 @@ BOOST_FIXTURE_TEST_CASE(ProductivityStats, WorldFixtureEmpty1P)
     BOOST_TEST(buildingRegister.CalcProductivities() == expectedProductivity, per_element());
     BOOST_TEST(buildingRegister.CalcAverageProductivity() == 0u);
 
+    std::vector<BuildingType> buildingTypes;
+    const auto buildingTypesEnum = helpers::EnumRange<BuildingType>{};
+
+    for(auto it = buildingTypesEnum.begin(); it != buildingTypesEnum.end(); ++it)
+    {
+        buildingTypes.push_back(*it);
+    }
```

Comment:
We use C++11/14 so you never need this `for(it=..., ++it)` but can always use the range-based-for: `for(auto t: buildingTypesEnum)`

However here you can just use the range constructor:
```suggestion
    const auto buildingTypesEnum = helpers::EnumRange<BuildingType>{};
    std::vector<BuildingType> buildingTypes(buildingTypesEnum.begin(), buildingTypesEnum.end());
```

Reply:
I looked into this, and didnt manage to make it work. 

believe you have to implement https://en.cppreference.com/w/cpp/iterator/iterator_tags 

`E0289	no instance of constructor "std::vector<_Ty, _Alloc>::vector [with _Ty=BuildingType, _Alloc=std::allocator<BuildingType>]" matches the argument list	Test_integration	\s25client\tests\s25Main\integration\testGamePlayer.cpp	73		
`